### PR TITLE
Recover maxrows.test.js

### PR DESCRIPTION
**Pull request by jannyHou**
connect to strongloop/loopback-connector-cloudant#33

write request still runs into limit problem.

### CODE REVIEW

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

File: test/maxrows.test.js [hunk 1]
```diff
@@ -26,7 +25,7 @@ describe('cloudant max rows', function() {
     });
     Thing.belongsTo('foo', {model: Foo});
     Foo.hasMany('things', {foreignKey: 'fooId'});
-    db.automigrate(done);
+    done();
```
Comment (kjdelisle) [inline test/maxrows.test.js]: Was this changed because the db.automigrate call was wiping out data?

File: test/cloudant.test.js [hunk 2]
```diff
@@ -191,7 +191,7 @@ describe('cloudant connector', function() {
       CustomerSimple.create(seed(), done);
     });
     describe('missing in modelDef', function() {
-      it('returns result when nested property is not' +
+      it('returns result when nested property is not ' +
```
Comment (kjdelisle) [inline test/cloudant.test.js]: unnecessary concatenation, can be written as
```
    it('returns result when nested property is not an array type',
      function(done) {
```

File: test/maxrows.test.js (hunk 1 above)
Comment (jannyHou) [inline test/maxrows.test.js]: @kjdelisle there is another bug of `automigrate`: https://github.com/strongloop/loopback-connector-cloudant/issues/92

Could be related to https://github.com/strongloop/loopback-connector-cloudant/issues/94

File: test/cloudant.test.js (hunk 2 above)
Comment (kjdelisle) [inline test/cloudant.test.js]: Should still be under 80 lines, even when indented twice.

Comment (jannyHou) [inline test/cloudant.test.js]: it will exceed the line's max characters, that's why I have to break them down

**Comment by jannyHou** [comment]
Blocked by https://github.com/strongloop/loopback-connector-cloudant/issues/92, will get back to this pr after #92 fixed.

**Comment by jannyHou** [comment]
Created a pr to fix issue#92 https://github.com/strongloop/loopback-connector-cloudant/pull/102

**Comment by jannyHou** [comment]
I would prefer to use `()=>{}` than `function() {}`, but lamda function is not flexible to add a function name, any thought?

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

**Comment by jannyHou** [comment]
@slnode test please

File: test/maxrows.test.js [hunk 3]
```diff
@@ -26,7 +22,16 @@ describe('cloudant max rows', function() {
     });
     Thing.belongsTo('foo', {model: Foo});
     Foo.hasMany('things', {foreignKey: 'fooId'});
-    db.automigrate(done);
+    db.automigrate(function cleanUpData(err) {
+      if (err) return done(err);
+      Thing.destroyAll(function removeModelInstances(err) {
+        if (err) return done(err);
+        Foo.destroyAll(function removeModelInstances(err) {
```
Comment (rmg) [inline test/maxrows.test.js]: Giving these callbacks the same name partially defeats the purpose of naming them, doesn't it?

Comment (jannyHou) [inline test/maxrows.test.js]: In this case, true....the cb function name doesn't give much extra info than the api name `destroyAll` :p

**Pull request merged**